avs_response_code.go

```diff
@@ -0,0 +1,21 @@
+package braintree
+
+type AVSResponseCode string
+
+const (
+	AVSMatches      AVSResponseCode = "M" // The postal code or street address provided matches the information on file with the cardholder's bank.
```

Comment:
Part of me thinks we should name these fields with the prefix like the `GatewayRejectionReason` consts, so that would change these to be `AVSResponseCodeMatches`. It'll make it long for some, but I think the consistency will be beneficial. Thoughts?

cc @lionelbarrow